language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MatchClauseNodeContext.java

```diff
@@ -53,18 +53,31 @@ public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context,
             completionItems.addAll(this.getCompletionItemList(moduleContent, context));
         } else if (onSuggestIfClause(context, node)) {
             completionItems.add(new SnippetCompletionItem(context, Snippet.KW_IF.get()));
+        } else if (onSuggestionsAfterDoubleArrow(context, node)) {
+            /*
+            Handles the following
+            eg: match v {
+                    1 => <cursor>
+                }
+             */
+            return completionItems;
         } else {
             completionItems.addAll(this.getPatternClauseCompletions(context));
         }
         this.sort(context, node, completionItems);
 
         return completionItems;
     }
-    
+
+    private boolean onSuggestionsAfterDoubleArrow(BallerinaCompletionContext context, MatchClauseNode node) {
+        return context.getCursorPositionInTree() >= node.rightDoubleArrow().textRange().endOffset();
+    }
+
     private boolean onSuggestIfClause(BallerinaCompletionContext context, MatchClauseNode node) {
         int cursor = context.getCursorPositionInTree();
         SeparatedNodeList<Node> matchPatterns = node.matchPatterns();
         
-        return !matchPatterns.isEmpty() && cursor > matchPatterns.get(matchPatterns.size() - 1).textRange().endOffset();
+        return !matchPatterns.isEmpty() && cursor > matchPatterns.get(matchPatterns.size() - 1).textRange().endOffset()
+                && cursor <= node.rightDoubleArrow().textRange().startOffset();
```

Comment:
Let's verify the presence of the `rightDoubleArrow`

Reply:
Done 👍